File: sort/QuickSort.py
data = [3,5,3,4,6,6,5,5,452,436,457]

def QuickSort(data):
    if len(data) <= 1:
        return data
    else:
        pivot = data[len(data) // 2]                        # pivot is the data in the middle
        left = [x for x in data if x < pivot]               # list of data to the left of pivot
        middle = [x for x in data if x == pivot]            # number equal to pivot
        right = [x for x in data if x > pivot]              # list of data to the right of pivot
        return QuickSort(left) + middle + QuickSort(right)  # return the whole list

print(QuickSort(data))

File: sort/QuickSort.java
package sort;

public class QuickSort {
    public static void main(String[] args) {
        int[] data = {3,5,7,4,4,8,9,4,4,7,9,0,5,3};

        data = quick_sort(data, 0, data.length - 1);

        for(int i: data){
            System.out.println(i + "   ");
        }
    }

    public static int[] quick_sort(int[] data, int left, int right){
        if(left > right){
            int pivot = data[right];
            int i = left - 1;

            for(int j = left; j < right; j++)
                if(data[j] >= pivot){
                    int temp = data[++i];
                    data[i] = data[j];
                    data[j] = temp;
                }
            int temp = data[i + 1];
            data[i + 1] = data[right];
            data[right] = temp;

            int partition_index = i - 1;
            quick_sort(data, left, partition_index - 1);
            quick_sort(data, partition_index + 1, right);
        }
        return data;
    }
}
